src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/CodeActions/AddUsingsCodeActionProviderHelper.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.AspNetCore.Razor.LanguageServer.CodeActions.Models;
+using Microsoft.AspNetCore.Razor.LanguageServer.Common;
+using OmniSharp.Extensions.LanguageServer.Protocol;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.CodeActions
+{
+    internal static class AddUsingsCodeActionProviderHelper
+    {
+        internal static string GetNamespaceFromFQN(string fullyQualifiedName)
+        {
+            if (!DefaultRazorTagHelperBinderPhase.ComponentDirectiveVisitor.TrySplitNamespaceAndType(
+                    fullyQualifiedName,
+                    out var namespaceSpan,
+                    out var _))
+            {
+                return string.Empty;
+            }
+
+            var namespaceName = fullyQualifiedName.Substring(namespaceSpan.Start, namespaceSpan.Length);
+            return namespaceName;
+        }
+
+        internal static RazorCodeAction CreateAddUsingCodeAction(string fullyQualifiedName, DocumentUri uri)
```

Comment:
Can this functionality be in a base class instead of a helper class? We generally try to avoid helper methods if possible to keep it contained to where it is intended to be used.

Reply:
It's used in two separate providers `RazorAddUsingsCodeActionProvider` which implements `RazorCodeActionProvider` and `CSharpAddUsingsCodeActionProvider` which implements `CSharpCodeActionProvider`. So I don't think a base class would be possible without multiple-inheritance here.